libpod/config/config.go

```diff
@@ -0,0 +1,607 @@
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	// _defaultTransport is a prefix that we apply to an image name to check
+	// docker hub first for the image.
+	_defaultTransport = "docker://"
+
+	// _rootlessConfigPath is the path to the rootless libpod.conf in $HOME.
+	_rootlessConfigPath = ".config/containers/libpod.conf"
+
+	// _conmonMinMajorVersion is the major version required for conmon.
+	_conmonMinMajorVersion = 2
+
+	// _conmonMinMinorVersion is the minor version required for conmon.
+	_conmonMinMinorVersion = 0
+
+	// _conmonMinPatchVersion is the sub-minor version required for conmon.
+	_conmonMinPatchVersion = 1
+
+	// _conmonVersionFormatErr is used when the expected versio-format of conmon
+	// has changed.
+	_conmonVersionFormatErr = "conmon version changed format"
+
+	// _cgroupV2FallbackToCrun can be set at compile-time and allows for falling
+	// back to using crun when running on system with cgroupsv2. This is turned
+	// on by default and only used as a precaution - some distributions might
+	// not want it.
+	_cgroupV2FallbackToCrun = true
+)
+
+var (
+	// InstallPrefix is the prefix where podman will be installed.
+	// It can be overridden at build time.
+	installPrefix = "/usr"
+
+	// EtcDir is the sysconfdir where podman should look for system config files.
+	// It can be overridden at build time.
+	etcDir = "/etc"
+
+	// SeccompDefaultPath defines the default seccomp path.
+	SeccompDefaultPath = installPrefix + "/share/containers/seccomp.json"
+
+	// SeccompOverridePath if this exists it overrides the default seccomp path.
+	SeccompOverridePath = etcDir + "/crio/seccomp.json"
+
+	// _rootConfigPath is the path to the libpod configuration file
+	// This file is loaded to replace the builtin default config before
+	// runtime options (e.g. WithStorageConfig) are applied.
+	// If it is not present, the builtin default config is used instead
+	// This path can be overridden when the runtime is created by using
+	// NewRuntimeFromConfig() instead of NewRuntime().
+	_rootConfigPath = installPrefix + "/share/containers/libpod.conf"
+
+	// _rootOverrideConfigPath is the path to an override for the default libpod
+	// configuration file. If OverrideConfigPath exists, it will be used in
+	// place of the configuration file pointed to by ConfigPath.
+	_rootOverrideConfigPath = etcDir + "/containers/libpod.conf"
+)
+
+// Config contains configuration options used to set up a libpod runtime
+type Config struct {
+	// NOTE: when changing this struct, make sure to update (*Config).Merge().
+
+	// StateType is the type of the backing state store. Avoid using multiple
+	// values for this with the same containers/storage configuration on the
+	// same system. Different state types do not interact, and each will see a
+	// separate set of containers, which may cause conflicts in
+	// containers/storage. As such this is not exposed via the config file.
+	//
+	// TODO: this is runtime data and doesn't belong here.
```

Comment:
Eh? I do see this as part of config. If we ever added a SQL store, I'd want to be able to configure Podman to use it via libpod.conf